set_user.c

```diff
@@ -149,6 +152,47 @@ extern Datum set_user(PG_FUNCTION_ARGS);
 void _PG_init(void);
 void _PG_fini(void);
 
+/*
+ * check_user_whitelist
+ *
+ * Check if user is contained by whitelist
+ *
+ */
+static bool
+check_user_whitelist(Oid userId, const char *whitelist)
+{
+	char	   *rawstring = NULL;
+	List	   *elemlist;
+	ListCell   *l;
+
+	if (whitelist == NULL || whitelist[0] == '\0')
+		return false;
+
+	rawstring = pstrdup(whitelist);
+
+	/* Parse string into list of identifiers */
+	if (!SplitIdentifierString(rawstring, ',', &elemlist))
+	{
+		/* syntax error in list */
+		ereport(ERROR,
+				(errcode(ERRCODE_SYNTAX_ERROR),
+				 errmsg("Invalid syntax in parameter")));
+	}
+
+	/* See if the whitelist contains the current username or wildcard. */
+	foreach(l, elemlist)
```

Comment:
I don't think we want to allow multiple elements if one of them is the wildcard character. Basically if there is a wildcard it ought to be the only element in the list. Anything else ought to throw an ERROR.